### DISCUSSION TITLE
How do i set up the simple login?

### DISCUSSION THREAD

**Discussion by lontraeye** [eclipse-oct/open-collaboration-tools#77] [Q&A]
Sorry for my noob question 😃

I've set up my Docker, but when I try to create a session, it redirects me to a login page. What is this login page, and how can I create or dismiss the authentication?

**Comment by msujew** [comment]
Hey @lontraeye,

the default login page enabled with the `OCT_ACTIVATE_SIMPLE_LOGIN` variable is there for users to set their name (i.e. to create non-persisted profile). It's not a real login mechanism. Joining a session without creating a profile is not possible, as users should have at least a name associated with them. Therefore it also cannot be skipped/dismissed. You can also add real auth provider (right now we only support GitHub and Google) via env variables. See:

https://github.com/TypeFox/open-collaboration-tools/blob/86f2afd71aa2b935f9cbdecc969fcbc2ba4f9cf0/packages/open-collaboration-server/src/auth-endpoints/oauth-endpoint.ts#L122-L124

Eventually, we want to use the server API directly in the VS Code extension to log in anonymously (only set a name), but we're not quite there yet.

**Reply by lontraeye** [reply]
So the this option is there for me to build an auth process?

Its easier to just use github auth then?

**Reply by msujew** [reply]
> So the this option is there for me to build an auth process?

The option is there for people hosting a OCT server who don't care about unauthorized people accessing their service (for example the people using the publicly hosted instance of this server).

> Its easier to just use github auth then?

I don't really get the question. Both authentication mechanisms are pretty simple - one makes use of GitHub, the other allows users to enter any name they want other users to see.

**Reply by lontraeye** [reply]
I am supposed to just put my name there and send? cause its causing an error on the endpoint (Sorry if i sound mad or rude, English is not my main language)

**Reply by msujew** [reply]
Right, as a security mechanism, the endpoint is only available if yoi set the `OCT_ACTIVATE_SIMPLE_LOGIN` environment variable to `true`.
